utils/release-candidate-testing/python/rc_test.py

```diff
@@ -0,0 +1,206 @@
+    array_of_keys: List[bytes] = [f"foo{i}".encode() for i in range(1, 4)]
+    # delete the keys
+    deleted_keys_num = await client.delete(array_of_keys)
+    print(f"Deleted keys: {deleted_keys_num}")
+    # check that the correct number of keys were deleted
+    if deleted_keys_num != 3:
+        print(deleted_keys_num)
+        raise Exception(
+            f"Unexpected number of keys deleted, expected 3, got {deleted_keys_num}"
+        )
+    # check that the keys were deleted
+    for i in range(1, 4):
+        val = await client.get(f"foo{i}")
+        if val is not None:
+            print(val)
+            raise Exception(f"Unexpected value, expected None, got {val}")
+    print("Keys deleted successfully")
+
+    # Test INCR command
+    incr_key = b"counter"
+    await client.set(incr_key, b"0")
+    incr_result = await client.incr(incr_key)
+    if incr_result != 1:
+        raise Exception(f"Unexpected INCR result, expected 1, got {incr_result}")
+    print("INCR command successful")
+
+    # Test LPUSH and LRANGE commands
+    list_key = b"mylist"
+    await client.lpush(list_key, [b"world", b"hello"])
+    list_values = await client.lrange(list_key, 0, -1)
+    if list_values != [b"hello", b"world"]:
+        raise Exception(
+            f"Unexpected LRANGE result, expected [b'hello', b'world'], got {list_values}"
+        )
+    print("LPUSH and LRANGE commands successful")
+
+    # Test HSET and HGETALL commands
+    hash_key = b"myhash"
+    await client.hset(hash_key, {b"field1": b"value1", b"field2": b"value2"})
+    hash_values = await client.hgetall(hash_key)
+    if hash_values != {b"field1": b"value1", b"field2": b"value2"}:
+        raise Exception(
+            f"Unexpected HGETALL result, expected {{b'field1': b'value1', b'field2': b'value2'}}, got {hash_values}"
+        )
+    print("HSET and HGETALL commands successful")
+
+    print("All commands executed successfully")
+
+
+async def create_cluster_client(
+    nodes_list: List[NodeAddress] = [("localhost", 6379)]
+) -> GlideClusterClient:
+    addresses: List[NodeAddress] = nodes_list
+    config = GlideClusterClientConfiguration(
+        addresses=addresses,
+        client_name="test_cluster_client",
+    )
+    client = await GlideClusterClient.create(config)
+    print("Cluster client created successfully")
+    return client
+
+
+async def create_standalone_client(server: List[NodeAddress]) -> GlideClient:
+    config = GlideClientConfiguration(
+        addresses=server,
+        client_name="test_standalone_client",
+    )
+    client = await GlideClient.create(config)
+    print("Standalone client created successfully")
+    return client
+
+
+async def test_standalone_client() -> None:
+    print("Testing standalone client")
+    output = start_servers(False, 1, 1)
+    servers, folder = parse_cluster_script_start_output(output)
+    standalone_client = await create_standalone_client(servers)
+    await run_commands(standalone_client)
```

Comment:
Run all IT instead

Reply:
We already had this discussion with Bar on the node rc tests.
That's not what those tests should do.
We have IT tests for a reason, and we have rc tests for another, the goal is just a slight check that nothing weird happened, mainly the compilation and packing.
Not to test all functionality.

Reply:
We can't find missing exports with these tests. We found missing exports for node client just because I ran all IT on an RC (#2307).
For basic checks you can run benchmark - and again without re-inventing a wheel.

Reply:
If we want to run IT against the packed packages we should do it regularly using the npm pack for example and use it as "artifact".
Testing it during release doesn't make sense.

Reply:
I understand that it might be needed, but not during release, not against all host types, and do it regularly locally using packing.

The "not inventing the wheel" in the case of using the benchmarks is really much more pain than gain.
You take three different components of the repo with three different use cases and try to jingle between them when python anyway doesn't like stuff that is not in its environment.

I think its an over engineering.

Reply:
https://github.com/valkey-io/valkey-glide/issues/2316#issuecomment-2359464859